### ISSUE TITLE
Weird failures using sets

### ISSUE THREAD

**Issue by Zac-HD** [pschanely/CrossHair#297]
**Expected vs actual behavior**
The code below runs without error in CPython, and fails with `TypeError: BadCompare.__eq__() takes 1 positional argument but 2 were given` under Crosshair.

(I was originally debugging a `RecursionError` using https://github.com/DRMacIver/shrinkray, but my test script was only conditioned on 'fails on crosshair, passes on hypothesis backend' and so here we are.  I'll make another attempt with a more specific condition too.)

[**To Reproduce**](https://crosshair-web.org/?crosshair=0.1&python=3.8&gist=ca96a666b4f010a8d065834dd82fed6c)
```python
class BadCompare:
    def __eq__(self):
        pass

    def __hash__(self):
        return 0

def test_repro(x: bool):
    """post: True"""
    for _ in set([BadCompare]) | {BadCompare()}:
        pass
```

**Comment by Zac-HD** [comment]
Turns out that the `RecursionError` is even more trivial - it doesn't even need any elements in the set!

```python
def test_repro(x: bool):
    """post: True"""
    s = set()
    copy = set(s)
    s |= s
    s != copy
```

**Comment by pschanely** [comment]
Thanks!!!

The first case is tricky. Was it intentional to not instantiate BadCompare in `set([BadCompare])`? The type is highly unlikely to hash to zero, so I don't think normal Python ever attempts the equality check. CrossHair however, avoids hashing and jumps to an exploding equality comparison. So far, my gut says to accept this unsoundness, and encourage people to use other tests to ensure their `__eq__` and `__hash__` methods are well-formed, pure, and [consistent with each other](https://github.com/pschanely/CrossHair/blob/main/crosshair/examples/PEP316/bugs_detected/hash_consistent_with_equals.py).

The second case is a plain old bug, and an important one! Fix is in progress.

**Comment by Zac-HD** [comment]
The first case was just what `shrinkray` gave me when I wasn't careful enough about the condition ([reducers are fuzzers](https://blog.regehr.org/archives/1284)!) - I don't have any real reason to care about it, but figured I might as well report that too while the second one was processing 🙂

**Comment by pschanely** [comment]
> The first case was just what `shrinkray` gave me when I wasn't careful enough about the condition ([reducers are fuzzers](https://blog.regehr.org/archives/1284)!) - I don't have any real reason to care about it, but figured I might as well report that too while the second one was processing 🙂

Got it! I've hopefully fixed the set-related recursion errors in 0.0.67. And I've started a discussion about the larger question brought up by your first example in #299
